reconcile/sentry_config.py

```diff
@@ -0,0 +1,317 @@
+import logging
+import reconcile.queries as queries
+import utils.gql as gql
+import utils.secret_reader as secret_reader
+
+from reconcile.github_users import init_github
+from utils.config import get_config
+from utils.sentry_client import SentryClient
+
+
+SENTRY_PROJECTS_QUERY = """
+{
+  apps: apps_v1 {
+    sentryProjects {
+      team {
+        name
+      }
+      projects {
+        name
+        description
+        email_prefix
+        platform
+        sensitive_fields
+        safe_fields
+        auto_resolve_age
+        allowed_domains
+      }
+    }
+  }
+}
+"""
+
+SENTRY_USERS_QUERY = """
+{
+  roles: roles_v1 {
+    name
+    users {
+      name
+      github_username
+    }
+    bots {
+      name
+      github_username
+    }
+    permissions {
+      service
+      ...on PermissionSentryTeam_v1 {
+        team
+      }
+    }
+  }
+}
+"""
+
+SENTRY_TEAMS_QUERY = """
+{
+  teams: sentry_teams_v1 {
+    name
+  }
+}
+"""
+
+class SentryState:
+  def __init__(self):
+    # Map of user:teams[]
+    self.users = {}
+    # List of team names
+    self.teams = []
+    # Map of team:projects_config[]
+    self.projects = {}
+  
+  def init_users_from_current_state(self, users):
+    self.users = users
+
+  def init_users_from_desired_state(self, users):
+    # Input is in the form of team:members[]
+    for team in users.keys():
+      for user in users[team]:
+        if team not in self.users.keys():
+          self.users[user] = [team]
+        else:
+          self.users[user].append(team)
+
+  def init_projects_from_current_state(self, client, projects):
+    # Input is in the form of project:teams[]
+    for project in projects.keys():
+      for team in projects[project]:
+        # TODO: Retrieve project and store relevant config
+        p = client.get_project(project)
+        pdata = {
+          "name": p["name"],
+          "email_prefix": p["subjectPrefix"],
+          "platform": p["platform"]
+        }
+        if "sensitiveFields" in p.keys():
+          pdata["sensitive_fields"] = p["sensitiveFields"]
+        if "safeFields" in p.keys():
+          pdata["safe_fields"] = p["safeFields"]
+        if "resolveAge" in p.keys():
+          pdata["auto_resolve_age"] = p["resolveAge"]
+        if "allowedDomains" in p.keys():
+          pdata["allowed_domains"] = p["allowedDomains"]
```

Comment:
super minor comment, but there's an opportunity here to restructure and refactor this with a for loop and a single if

Reply:
Done